### PR TITLE
Translate to es6

### PR DESCRIPTION
**Pull request by IuriiBod**

### CODE REVIEW

File: lib/both.js [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+        subQuerys = this._buildQuerysByFields(configEntry.realetive.fields, escapedSearchString);
+      }
+      // searchStringQueries = configEntry.fields.map(function(fieldName) {
+      //   let $orEntry = {};
+      //   $orEntry[fieldName] = {
+      //     $regex: escapedSearchString,
+      //     $options: 'i'
+      //   };
+      //   return $orEntry;
+      // });
```
Comment (trsdln) [inline lib/both.js]: @IuriiBod we don't need commented code.

Comment (IuriiBod) [inline lib/both.js]: ok

**Comment by trsdln** [comment]
@IuriiBod Basically everything inside both.js should be on client side. Also, some things from AntiSearchSource can be moved to server side (like checker). Can you move everything to correct places?

**Comment by trsdln** [comment]
@IuriiBod we don't need both.js at all. It should be splitted between client and server.

File: lib/client.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Blaze.TemplateInstance.prototype.AntiSearchSource = (config) => {
+  return new AntiSearchSource._clientProto(config, this);
```
Comment (trsdln) [inline lib/client.js:2]: @IuriiBod after removing both.js AntiSearchSourceClient may be used directly.

File: lib/server.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    maxLimit: Match.Optional(Number),
+    securityCheck: Match.Optional(Function),
+    allowedFields: Match.Optional([String]),
+    queryTransform: Match.Optional(Function)
+  });
+  this._allowRules[collectionName] = allowRules;
+  return this._allowRules[collectionName];
+};
+
+AntiSearchSource._SearchConfig = Match.Where(function(config) {
```
Comment (trsdln) [inline lib/server.js:12]: @IuriiBod this is redundant. May be defined as local cost variable or used directly inside method.

File: lib/both.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+}
+
+
+this.AntiSearchSource = {
+  _allowRules: {},
+  _clientProto: AntiSearchSourceClient,
+  _escapeRegExpStr: function(str) {
+    return str.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&");
+  },
+  _buildSearchQuery: function(configEntry) {
```
Comment (trsdln) [inline lib/both.js:133]: @IuriiBod Looks like this method should stay in both.

File: lib/both.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+    } else {
+      return this._searchResult;
+    }
+  }
+
+}
+
+
+this.AntiSearchSource = {
+  _allowRules: {},
```
Comment (trsdln) [inline lib/both.js:128]: @IuriiBod to server

File: lib/both.js [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+  _clientProto: AntiSearchSourceClient,
```
Comment (trsdln) [inline lib/both.js:129]: don't need at all

File: lib/both.js [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+  _escapeRegExpStr: function(str) {
+    return str.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&");
+  },
```
Comment (trsdln) [inline lib/both.js:132]: stays here

File: lib/both.js [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+      return this._collection.find(query, options);
+    } else {
+      return this._searchResult;
+    }
+  }
+
+}
+
+
+this.AntiSearchSource = {
```
Comment (trsdln) [inline lib/both.js:127]: this - redundant. Simple global variable is fine.

File: lib/both.js [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+    if (_.isArray(subQuery)) {
+      let len = searchQuery.$and.length -1;
+      for (let q of subQuery) {
+        searchQuery.$and[len].$or.push(q);
+      }
+    }
+
+    return searchQuery;
+  },
+  create: function(config) {
```
Comment (trsdln) [inline lib/both.js:182]: redundant.

File: lib/both.js [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+      return {[fieldName]: {
+          $regex: escapedSearchString,
+          $options: 'i'
+        }
+      };
+    });
+
+    return result;
+  },
+  _addInSearchQuery: function(query, subQuery) {
```
Comment (trsdln) [inline lib/both.js:171]: @IuriiBod should be renamed or better documented.

File: lib/both.js [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+    }
+
+    return {
+      searchQuery: searchQuery,
+      subQuerys: subQuerys
+    };
+  },
+  _buildQuerysByFields: function(fields, escapedSearchString) {
+    let result = fields.map(function(fieldName) {
+      return {[fieldName]: {
```
Comment (trsdln) [inline lib/both.js:162]: hard to read. Multiline object should have only "{" at first line. All properties are listed from new line.

File: lib/server.js [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+    configEntry.fields.forEach(function(fieldName) {
+      fieldsToPublish[fieldName] = 1;
+      return fieldsToPublish[fieldName];
+    });
+
+    return collection.find(searchQuery, {
+      limit: configEntry.limit,
+      fields: fieldsToPublish
+    }).fetch();
+  }
```
Comment (trsdln) [inline lib/server.js:91]: this method is too big and should be refactored.

**Comment by trsdln** [comment]
@IuriiBod I clarified what I mean under "refactoring".

**Pull request closed without merging**